src/codeflare_sdk/cluster/config.py

```diff
@@ -63,6 +80,46 @@ def __post_init__(self):
             )
         self._memory_to_string()
         self._str_mem_no_unit_add_GB()
+        self._gpu_to_resource()
+        self._combine_custom_resource_mapping()
+
+    def _combine_custom_resource_mapping(self):
+        if overwritten := set(self.custom_resource_mapping.keys()).intersection(
+            DEFAULT_RESOURCE_MAPPING.keys()
+        ):
+            if self.overwrite_default_resource_mapping:
+                warnings.warn(
+                    f"Overwriting default resource mapping for {overwritten}",
+                    UserWarning,
+                )
+            else:
+                raise ValueError(
+                    f"Resource mapping already exists for {overwritten}, set overwrite_default_resource_mapping to True to overwrite"
+                )
+        self.custom_resource_mapping = {
+            **DEFAULT_RESOURCE_MAPPING,
+            **self.custom_resource_mapping,
+        }
+
+    def _gpu_to_resource(self):
+        if self.head_gpus:
+            warnings.warn(
+                "head_gpus is being deprecated, use head_custom_resource_requests"
```

Comment:
Maybe this warning message could also give an example of how to specify the resources since it isn't a simple changeover of the variable name but also a change in type.